src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/ManagedIndexRunner.kt

```diff
@@ -146,21 +145,45 @@ object ManagedIndexRunner : ScheduledJobRunner,
         val action: Action? = state?.getActionToExecute(clusterService, client, managedIndexMetaData)
         val step: Step? = action?.getStepToExecute()
 
-        // If any of the above components come back as null then we are moving to error in ManagedIndexMetaData
-        if (state != null && action != null && step != null) {
-            step.execute()
+        if (action?.config?.configRetry?.backoff?.shouldBackoff(managedIndexMetaData.actionMetaData, action.config.configRetry) == true) {
+            // If we should back off then exit early.
+            logger.info("Backoff for retrying.")
+            return
         }
 
-        val updatedManagedIndexMetaData = managedIndexMetaData.getUpdatedManagedIndexMetaData(state, action, step)
-
-        // TODO: Check if we can move this into the TransportUpdateManagedIndexMetaDataAction to cover all cases where
-        //  IndexMetaData does not exist anymore since if this current execution was a delete step and it was
-        //  successful then the IndexMetaData will be wiped and will throw a NPE if we attempt to update it
-        if (updatedManagedIndexMetaData.isSuccessfulDelete) {
+        // If Step status is still in Start it means we have failed to update the IndexMetaData.
+        // TODO: In case the step is Idempotent we can retry. ie. open, close, read_only, read_write, etc...
+        if (managedIndexMetaData.stepMetaData?.stepStatus == Step.StepStatus.STARTING) {
+            val info = mapOf("message" to "Previous action was not able to update IndexMetaData.")
+            updateManagedIndexMetaData(managedIndexMetaData.copy(policyRetryInfo = PolicyRetryInfoMetaData(true, 0), info = info))
+            disableManagedIndexConfig(managedIndexConfig)
             return
         }
 
-        updateManagedIndexMetaData(updatedManagedIndexMetaData)
+        // If any of the above components come back as null then we are moving to error in ManagedIndexMetaData
+        val startingManagedIndexMetaData = managedIndexMetaData.getStartingManagedIndexMetaData(state, action, step)
+        updateManagedIndexMetaData(startingManagedIndexMetaData)
```

Comment:
I changed so that the updateManagedIndexMetaData after the `step.execute` uses this `startingManagedIndexMetaData` so it will have second chance at updating the metadata. 

If `state`, `action`, and `step` are null and we don't run `step.execute`, bad things can happen. :(